shell/browser/native_window_views.cc

```diff
@@ -1932,4 +1940,46 @@ std::unique_ptr<NativeWindow> NativeWindow::Create(
   return std::make_unique<NativeWindowViews>(options, parent);
 }
 
+#if BUILDFLAG(IS_WIN)
+bool NativeWindowViews::PreHandleMSG(UINT message,
+                                     WPARAM w_param,
+                                     LPARAM l_param,
+                                     LRESULT* result) {
+  if (message == WM_DPICHANGED) {
+    WORD new_dpi_y = LOWORD(w_param);
+    RECT* const prcNewWindow = reinterpret_cast<RECT*>(l_param);
+
+    ::SetWindowPos(GetAcceleratedWidget(),
+                   nullptr,
+                   prcNewWindow->left,
+                   prcNewWindow->top,
+                   prcNewWindow->right - prcNewWindow->left,
+                   prcNewWindow->bottom - prcNewWindow->top,
+                   SWP_NOZORDER | SWP_NOACTIVATE);
+
+    OnHostDpiChanged(new_dpi_y);
+    *result = 0;
+    return true;
+  }
+  // TODO(bridiver) handle other messages here
+  return false;
+}
+
+void NativeWindowViews::OnHostDpiChanged(int new_dpi) {
+  current_dpi_ = new_dpi;
+
+  if (widget()) {
+    widget()->UpdateWindowIcon();
+
+    if (widget()->non_client_view() && widget()->non_client_view()->frame_view()) {
+      widget()->non_client_view()->frame_view()->InvalidateLayout();
+      widget()->non_client_view()->frame_view()->SchedulePaint();
+    }
+
+    widget()->GetRootView()->Layout();
+    widget()->GetRootView()->SchedulePaint();
+  }
+}
```

Comment:
This seems to duplicate what the default handing is performed in the runtime today via https://source.chromium.org/chromium/chromium/src/+/main:ui/views/win/hwnd_message_handler.cc;l=1876-1921 , can you clarify why this is needed ?